nixos/modules/services/video/epgstation/default.nix

```diff
@@ -0,0 +1,297 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.epgstation;
+
+  username = config.users.users.epgstation.name;
+  groupname = config.users.users.epgstation.group;
+
+  settingsFmt = pkgs.formats.json {};
+  settingsTemplate = settingsFmt.generate "config.json" cfg.settings;
+  preStartScript = let
+    readPassword = f: if (f != null)
```

Comment:
Would it be valid for the password to be `null`? Why not make it a hard requirement in the first place?

Reply:
Password protection is an optional feature. Since it is generally not recommended to expose this service outside of your own network, it's valid not to have one.